config/checkstyle-checks.xml

```diff
@@ -46,6 +46,13 @@
     <property name="files"
               value=".*[\\/]src[\\/](test|it|xdocs-examples)[\\/].*(?&lt;!Support)\.java"/>
   </module>
+  <!-- require for charset testing -->
+  <module name="SuppressionSingleFilter">
+    <property name="checks" value="RegexpSinglelineJava"/>
+    <property name="files"
+              value=".*[\\/]src[\\/]test[\\/]java[\\/].*[\\/]RegexpHeaderCheckTest.java"/>
+    <property name="lines" value="398"/>
```

Comment:
Suppression by line is very fragile.
Please share link to check that forcing you to do this.

Is it https://github.com/checkstyle/checkstyle/blob/42f9ceb39a5e4b79e2733c9cedc1766fcafd5cbb/config/checkstyle-checks.xml#L827 ?

If so , use id of Check, not name.
Please use xpath suppression to suppress violation on special this method

Reply:
@romani does xpath query generator works only when a violation message contains column no 
https://github.com/checkstyle/checkstyle/blob/34e7d5c714b2df62781790537c6127b5712ec097/src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java#L111-L112
https://github.com/checkstyle/checkstyle/blob/34e7d5c714b2df62781790537c6127b5712ec097/src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java#L126-L127

To generate a query

Reply:
I have removed line no. I think Xpath Query may not be possible because this error i on line not on any ast token

Reply:
Please use suppression like https://github.com/checkstyle/checkstyle/blob/19b2594c7766801859e73f2194b55fb00d26f36c/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L819

Reply:
done